app/assets/javascripts/course/components/FormComponent.js

```diff
@@ -8,13 +8,22 @@ class FormComponent extends React.Component {
 
     this.state = {
       formCode: '',
-      error: ''
+      error: '',
+      success: '',
     }
 
     this.updateCodeState = this.updateCodeState.bind(this)
     this.formSubmit = this.formSubmit.bind(this)
   }
 
+  componentWillReceiveProps(nextProps) {
+    this.props.component != nextProps.component ? this.setState({
+                                                    formCode: '',
+                                                    error: '',
+                                                    success: '',
+                                                  })
+  }
+
```

Comment:
I think you're missing the else clause after ' : '. Does it compile on yours? Also I think this would be cleaner just broken up into if/else statements.

<img width="671" alt="screen shot 2017-02-20 at 3 18 47 pm" src="https://cloud.githubusercontent.com/assets/13477279/23145013/501f4676-f780-11e6-94e3-c6b2a710338b.png">